### PR TITLE
Codespell: adding config, workflow and fixing some typos

### PR DESCRIPTION
**Pull request by yarikoptic**
In the excitement of coming up hackathon and also since https://github.com/datalad/datalad-extension-template/pull/45 has been debated for nearly half a year, decided to make at least this one a little more a better spelled one ;)

### CODE REVIEW

**Comment by yarikoptic** [comment]
I don't see how these type fixes could result in appveyor fails like

```

>           raise AssertionError("dicts differ:\n%s" % "\n".join(msgs))
E           AssertionError: dicts differ:
E            ['root_dataset_id'] differs: '00010203-1011-2021-3031-404142434445' != '<unknown>'
../../../dlvenv/lib/python3.8/site-packages/datalad/tests/utils_pytest.py:1271: AssertionError
```

**Comment by adswa** [comment]
I have seen these failures also in core in the metalad tests, e.g. in https://github.com/datalad/datalad/pull/7308

**Comment by yarikoptic** [comment]
heh -- codespell is still failing due to fresh? `./docs/source/design/history.rst:75: formate ==> format`, will tune up for that now

File: datalad_metalad/filters/base.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def filter(self,
 
         This method is called by the 'meta-filter' driver. It should iterate
         through the metadata instances that are provided by the metadata
-        coordinates, perform the filter operation ond yield the resulting
+        coordinates, perform the filter operation one yield the resulting
```
Comment (adswa) [inline datalad_metalad/filters/base.py:41]:
```suggestion
        coordinates, perform the filter operation and yield the resulting
```

**Pull request merged**